pkg/template/apis/template/types.go

```diff
@@ -143,13 +188,26 @@ type TemplateInstanceConditionType string
 
 const (
 	// TemplateInstanceReady indicates the readiness of the template
-	// instantiation.
+	// instantiation or one of its objects.
 	TemplateInstanceReady TemplateInstanceConditionType = "Ready"
 	// TemplateInstanceInstantiateFailure indicates the failure of the template
-	// instantiation
+	// instantiation or one of its objects.
 	TemplateInstanceInstantiateFailure TemplateInstanceConditionType = "InstantiateFailure"
+	// TemplateInstanceWaiting indicates waiting for readiness or failure of the
+	// template instantiation or one of its objects.
+	TemplateInstanceWaiting TemplateInstanceConditionType = "Waiting"
 )
 
+// TemplateInstanceObject references an object created by a TemplateInstance.
+type TemplateInstanceObject struct {
+	//  Ref is a reference to the created object.
+	Ref kapi.ObjectReference
+
+	// Conditions represent the latest available observations of the object's
+	// current state.
+	Conditions []TemplateInstanceCondition
```

Comment:
Hmm, implementing this will be a fair amount of extra work because there are potentially many requirements per object and they'll have to be aggregated.  Not sure it's worth the effort.